actionpack/lib/abstract_controller/caching/fragments.rb

```diff
@@ -136,7 +136,7 @@ def expire_fragment(key, options = nil)
 
       def instrument_fragment_cache(name, key) # :nodoc:
         payload = instrument_payload(key)
-        ActiveSupport::Notifications.instrument("#{name}.#{instrument_name}", payload) { yield }
+        ActiveSupport::Notifications.instrument("#{name}.#{instrument_name}".freeze, payload) { yield }
```

Comment:
> But I think Ruby should also support this behavior?

No; it doesn't make any sense. `.freeze` isn't magical go-fast sprinkles: it tells the interpreter to do a particular thing -- store the string, and re-use it each time. If the string's going to have different contents each call, it's just not possible to do that.